CRM/Contribute/Form/SoftCredit.php

```diff
@@ -52,12 +52,6 @@ public static function preProcess(&$form) {
         'entity_id' => $contriDAO->contribution_page_id,
       );
       $profileId = CRM_Core_BAO_UFJoin::getUFGroupIds($ufJoinParams);
-
-      //check if any honree profile is enabled if yes then assign its profile type to $_honoreeProfileType
-      // which will be used to constraint soft-credit contact type in formRule, CRM-13981
-      if (!empty($profileId[0]) && !empty($profileId[2])) {
-        $form->_honoreeProfileType = CRM_Core_BAO_UFGroup::getContactType($profileId[0]);
-      }
```

Comment:
If you remove this part, I think we don't have any purpose to have preProcess() here as it seems this function was only used to set `_honoreeProfileType` in `$form`. Can we remove preProcess() from this file then ?